-1 
Tested at: 0a00eaa5423caa23727e61f85a7965a472e9f18c
 I found an error when building: 
 
 <pre>>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-26-1100/src/Utilities/StaticAnalyzers/src/EDMPluginDumper.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-26-1100/src/Utilities/StaticAnalyzers/src/FiniteMathChecker.cc 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-26-1100/src/Utilities/StaticAnalyzers/src/ArgSizeChecker.cpp 
>> Compiling  /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-26-1100/src/Utilities/StaticAnalyzers/src/CatchAll.cpp 
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-26-1100/src/FastSimulation/Tracking/plugins/TrajectorySeedProducer.cc: In constructor 'TrajectorySeedProducer::TrajectorySeedProducer(const edm::ParameterSet&)':
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-26-1100/src/FastSimulation/Tracking/plugins/TrajectorySeedProducer.cc:92:86: error: 'outerHit' was not declared in this scope
     measurementTrackerEvent->measurementTracker().geometricSearchTracker()->detLayer(outerHit.hit()->det()->geographicalId());
                                                                                      ^
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-26-1100/src/FastSimulation/Tracking/plugins/TrajectorySeedProducer.cc:93:81: error: 'innerHit' was not declared in this scope
   std::vector<BaseTrackerRecHit const *> innerHits(1,(const BaseTrackerRecHit*) innerHit.hit());
                                                                                 ^
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13051/10759/summary.html

--- a/FastSimulation/Tracking/plugins/TrajectorySeedProducer.cc
+++ b/FastSimulation/Tracking/plugins/TrajectorySeedProducer.cc
@@ -88,7 +88,25 @@ TrajectorySeedProducer::TrajectorySeedProducer(const edm::ParameterSet& conf):
         seedingLayers.push_back(std::move(trackingLayerList));
     }
 
-    // region producer
+       // region producer                                                                                                                                                  
+    if(conf.exists("RegionFactoryPSet")){
+        edm::ParameterSet regfactoryPSet = conf.getParameter<edm::ParameterSet>("RegionFactoryPSet");
+        std::string regfactoryName = regfactoryPSet.getParameter<std::string>("ComponentName");
+        theRegionProducer.reset(TrackingRegionProducerFactory::get()->create(regfactoryName,regfactoryPSet, consumesCollector()));
+
+        // seed creator                                                                                                                                                 
+        const edm::ParameterSet & seedCreatorPSet = conf.getParameter<edm::ParameterSet>("SeedCreatorPSet");
+        std::string seedCreatorName = seedCreatorPSet.getParameter<std::string>("ComponentName");
+        seedCreator.reset(SeedCreatorFactory::get()->create( seedCreatorName, seedCreatorPSet));
+    }
+}
+
+bool
+TrajectorySeedProducer::pass2HitsCuts(const TrajectorySeedHitCandidate & innerHit,const TrajectorySeedHitCandidate & outerHit) const
+{
+  const DetLayer * innerLayer =
+    measurementTrackerEvent->measurementTracker().geometricSearchTracker()->detLayer(innerHit.hit()->det()->geographicalId());
+  const DetLayer * outerLayer =
     measurementTrackerEvent->measurementTracker().geometricSearchTracker()->detLayer(outerHit.hit()->det()->geographicalId());
   std::vector<BaseTrackerRecHit const *> innerHits(1,(const BaseTrackerRecHit*) innerHit.hit());
   std::vector<BaseTrackerRecHit const *> outerHits(1,(const BaseTrackerRecHit*) outerHit.hit());

